cmd/commands/cmd_import_mission_control.go

```diff
@@ -98,3 +101,161 @@ func importMissionControl(ctx *cli.Context) error {
+		"option to shift timestamps. Note that this data is not " +
+		"persisted across restarts.",
+	Action: actionDecorator(loadMissionControl),
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "mcdatapath",
+			Usage: "The path to the querymc output file (json).",
+		},
+		cli.BoolFlag{
+			Name:  "discard",
+			Usage: "Discards current mission control data.",
+		},
+		cli.StringFlag{
+			Name: "timeoffset",
+			Usage: "Time offset to add to all timestamps. " +
+				"Format: 1m for 1 minute, 1h for 1 hour, 1d " +
+				"for 1 day.",
+		},
+	},
+}
+
+// loadMissionControl loads mission control data into an LND instance.
+func loadMissionControl(ctx *cli.Context) error {
+	rpcCtx := context.Background()
+
+	mcDataPath := ctx.String("mcdatapath")
+	if mcDataPath == "" {
+		return fmt.Errorf("mcdatapath must be set")
+	}
+
+	if _, err := os.Stat(mcDataPath); os.IsNotExist(err) {
+		return fmt.Errorf("%v does not exist", mcDataPath)
+	}
+
+	conn := getClientConn(ctx, false)
+	defer conn.Close()
+
+	client := routerrpc.NewRouterClient(conn)
+
+	// Load and unmarshal the querymc output file.
+	mcRaw, err := os.ReadFile(mcDataPath)
+	if err != nil {
+		return fmt.Errorf("could not read querymc output file: %w", err)
+	}
+
+	mc := &routerrpc.QueryMissionControlResponse{}
+	err = lnrpc.ProtoJSONUnmarshalOpts.Unmarshal(mcRaw, mc)
+	if err != nil {
+		return fmt.Errorf("could not unmarshal querymc output file: %w",
+			err)
+	}
+
+	// We discard mission control data if requested.
+	if ctx.Bool("discard") {
+		if !promptForConfirmation("This will discard mission control " +
+			"data: (yes/no)") {
+
+			return nil
+		}
+
+		_, err = client.ResetMissionControl(
+			rpcCtx, &routerrpc.ResetMissionControlRequest{},
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Add a time offset to all timestamps if requested.
+	timeOffset := ctx.String("timeoffset")
+	if timeOffset != "" {
+		offset, err := time.ParseDuration(timeOffset)
+		if err != nil {
+			return fmt.Errorf("could not parse time offset: %w",
+				err)
+		}
+
+		var maxTimestamp time.Time
+
+		for _, pair := range mc.Pairs {
+			if pair.History.SuccessTime != 0 {
+				unix := time.Unix(pair.History.SuccessTime, 0)
+				unix = unix.Add(offset)
+
+				if unix.After(maxTimestamp) {
+					maxTimestamp = unix
+				}
+
+				pair.History.SuccessTime = unix.Unix()
+			}
+
+			if pair.History.FailTime != 0 {
+				unix := time.Unix(pair.History.FailTime, 0)
+				unix = unix.Add(offset)
+
+				if unix.After(maxTimestamp) {
+					maxTimestamp = unix
+				}
+
+				pair.History.FailTime = unix.Unix()
+			}
+		}
+
+		fmt.Printf("Adding time offset %v to all timestamps. "+
+			"New max timestamp: %v\n", offset, maxTimestamp)
+	}
+
+	sanitizeMCData(mc.Pairs)
+
+	fmt.Printf("Mission control file contains %v pairs.\n", len(mc.Pairs))
+	if !promptForConfirmation("Import mission control data (yes/no):") {
+		return nil
+	}
+
+	_, err = client.XImportMissionControl(
+		rpcCtx,
+		&routerrpc.XImportMissionControlRequest{
```

Comment:
There's also the `Force` flag, not sure if that's useful to expose?

Reply:
Yes, good to have API feature completeness